File: Sources/Classes/RSApp.m
//
//  RSApp.m
//  RSSDKCore
//
//  Created by Arnab Pal on 17/10/19.
//  Copyright © 2019 RSlabs. All rights reserved.
//

#import "RSApp.h"

@implementation RSApp
- (instancetype)init
{
    self = [super init];
    if (self) {
        // Ref:  http://hboon.com/difference-between-cfbundleversion-and-cfbundleshortversionstring
        NSBundle *bundle = [NSBundle mainBundle];
        _build = [[bundle infoDictionary]valueForKey:@"CFBundleVersion"];
        _name = [[bundle infoDictionary]valueForKey:@"CFBundleName"];
        _nameSpace = [bundle bundleIdentifier];
        _version = [[bundle infoDictionary]valueForKey:@"CFBundleShortVersionString"];
    }
    return self;
}

- (NSDictionary<NSString *,NSObject *> *)dict {
    NSMutableDictionary *tempDict;
    @synchronized (tempDict) {
        tempDict = [[NSMutableDictionary alloc] init];

        [tempDict setValue:_build forKey:@"build"];
        [tempDict setValue:_name forKey:@"name"];
        [tempDict setValue:_nameSpace forKey:@"namespace"];
        [tempDict setValue:_version forKey:@"version"];

        return [tempDict copy];
    }
}
@end

File: Rudder/Rudder/Domain/Models/RSApp.swift
//
//  RSApp.swift
//  Rudder
//
//  Created by Desu Sai Venkat on 06/08/21.
//  Copyright © 2021 Rudder Labs India Pvt Ltd. All rights reserved.
//

import Foundation

struct RSApp {
    let build: String?
    let name: String?
    let nameSpace: String?
    let version: String?
    
    init(){
        let mainBundle = Bundle.main
        self.build = mainBundle.infoDictionary?["CFBundleVersion"] as? String
        self.name = mainBundle.infoDictionary?["CFBundleName"] as? String
        self.nameSpace = mainBundle.bundleIdentifier
        self.version = mainBundle.infoDictionary?["CFBundleShortVersionString"] as? String
    }
    
    func dict() -> [String:String]
    {
        var tempDict : [String:String] = [:]
        if(build != nil)
        {
            tempDict["build"] = build
        }
        if(name != nil)
        {
            tempDict["name"] = name
        }
        if(nameSpace != nil)
        {
            tempDict["namespace"] = nameSpace
        }
        if(version != nil)
        {
            tempDict["version"] = version
        }
        return tempDict
    }
}
